lib/MgiProxyStack.ts

```diff
@@ -0,0 +1,137 @@
+import * as cdk from '@aws-cdk/core';
+import {
+  addTags,
+  MilltechFXStack,
+  ResourceType,
+} from '@milltechfx/base-aws-stack';
+import * as lambda from '@aws-cdk/aws-lambda';
+import {Tracing} from '@aws-cdk/aws-lambda';
+import {EventBus} from '@aws-cdk/aws-events';
+import * as apigw from '@aws-cdk/aws-apigateway';
+import * as iam from '@aws-cdk/aws-iam';
+import path = require('path');
+
+interface MgiProxyStackProps extends cdk.StackProps {
+  systemEventBusArn: string;
+  stage: string;
+}
+
+export class MgiProxyStack extends MilltechFXStack {
+  lambdaLayer: lambda.LayerVersion;
+  api: apigw.RestApi;
+
+  constructor(scope: cdk.Construct, id: string, props: MgiProxyStackProps) {
+    super(scope, id, props);
+
+    addTags(this, {
+      'Tech:Service': 'MgiProxy',
+    });
+
+    const systemEventsBus = EventBus.fromEventBusArn(
+      this,
+      'SystemEventsBus',
+      props.systemEventBusArn
+    );
+
+    this.lambdaLayer = this.createLayer();
+
+    const cancelledTradeProxyLambda = this.createLambda(
+      'cancelledTradeProxy',
+      systemEventsBus.eventBusName
+    );
+
+    const orderStatusProxyLambda = this.createLambda(
+      'orderStatusProxy',
+      systemEventsBus.eventBusName
+    );
+
+    systemEventsBus.grantPutEventsTo(cancelledTradeProxyLambda);
+    systemEventsBus.grantPutEventsTo(orderStatusProxyLambda);
+
+    this.api = this.createApiGateway(props.stage);
+
+    this.addProxyLambdaIntegration(
+      cancelledTradeProxyLambda,
+      'CancelMgiExecutedTrade'
+    );
+
+    this.addProxyLambdaIntegration(orderStatusProxyLambda, 'ProxyMgiStatus');
+
+    this.createApiUserGroup();
+  }
+
+  createApiGateway(stage: string) {
+    return new apigw.RestApi(this, 'mgiProxy', {
+      deployOptions: {
+        stageName: stage,
+        dataTraceEnabled: true,
+      },
+      restApiName: this.getResourceName('mgiProxy', ResourceType.API_GATEWAY),
+      endpointTypes: [apigw.EndpointType.REGIONAL],
+    });
+  }
+
+  addProxyLambdaIntegration(
+    cancelledTradeProxyLambda: lambda.IFunction,
+    uri: string
+  ) {
+    const cancelledTradeProxyResource = this.api.root.addResource(uri);
+
+    cancelledTradeProxyResource.addMethod(
+      'POST',
+      new apigw.LambdaIntegration(cancelledTradeProxyLambda),
+      {
+        authorizationType: apigw.AuthorizationType.IAM,
+      }
+    );
+  }
+
+  createLayer(): lambda.LayerVersion {
+    const layerFiles = path.resolve(__dirname, '../assets/layer');
+    const lambdaLayer = new lambda.LayerVersion(this, 'mgiProxyLayer', {
+      code: lambda.Code.fromAsset(layerFiles),
+      compatibleRuntimes: [lambda.Runtime.NODEJS_12_X],
```

Comment:
Let use node runtime as fresh as possible
```suggestion
      compatibleRuntimes: [lambda.Runtime.NODEJS_14_X],
```